File: Graph/island/island.js
(function () {
  function DFS(edges, n, sv, visited) {
    visited[sv] = true;
    for (let i = 0; i < n; i++) {
      if (i == sv) continue;

      if (edges[sv][i] && !visited[i]) {
        DFS(edges, n, i, visited);
      }
    }
  }

  function island(edges, n) {
    const visited = [];
    for (let i = 0; i < n; i++) {
      visited[i] = false;
    }
    let cnt = 0;
    for (let i = 0; i < n; i++) {
      if (!visited[i]) {
        DFS(edges, n, i, visited);
        cnt++;
      }
    }
    return cnt;
  }

  (function main() {
    let v = 5;
    let e = 8;

    let connections = [
      { sv: 0, ev: 1 },
      { sv: 0, ev: 4 },
      { sv: 1, ev: 2 },
      { sv: 2, ev: 0 },
      { sv: 2, ev: 4 },
      { sv: 3, ev: 0 },
      { sv: 3, ev: 2 },
      { sv: 4, ev: 3 },
    ];

    let edges = [];
    for (let i = 0; i < v; i++) {
      edges[i] = [];
      for (let j = 0; j < v; j++) {
        edges[i][j] = false;
      }
    }

    for (let i = 0; i < e; i++) {
      let { sv, ev } = connections[i];
      edges[sv][ev] = true;
      edges[ev][sv] = true;
    }

    let ans = island(edges, v);
    console.log(ans);
  })();
})();

File: Graph/island/island.cpp
#include <iostream>
using std::cin;
using std::cout;
using std::endl;

void DFS(bool** edges, int& n, int& sv, bool* visited) {
    visited[sv] = true;
    for (int i{}; i < n; i++) {
        if (i == sv) continue;
        if (edges[sv][i] && !visited[i]) DFS(edges, n, i, visited);
    }
}

int island(bool** edges, int n) {
    bool* visited{new bool[n]};
    for (int i{}; i < n; i++) visited[i] = false;
    int cnt {};
    for (int i{}; i < n; i++) {
        if (!visited[i]) {
            DFS(edges, n, i, visited);
            cnt++;
        }
    }
    delete [] visited;
    return cnt;
}

int main() {
    int v, e;
    cin >> v >> e;

    bool** edges{new bool*[v]};
    for (int i{}; i < v; i++) {
        edges[i] = new bool[v];
        for (int j{}; j < v; j++) edges[i][j] = false;
    }

    for (int i{}; i < e; i++) {
        int sv, ev;
        cin >> sv >> ev;
        edges[sv][ev] = true;
        edges[ev][sv] = true;
    }

    int ans{island(edges, v)};
    cout << ans << endl;

    for (int i{}; i < v; i++) delete [] edges[i];
    delete [] edges;
}
